statannotations/statannotations.py

```diff
@@ -89,6 +91,54 @@ def find_x_position_box(b_plotter, box_name):
         box_pos = group_pos + hue_offset
         return box_pos
 
+    def get_xpos_location(pos, xranges):
+        """
+        Finds the x-axis location of a categorical variable
+        """
+        for xrange, _ in xranges.items():
+            if (pos >= xrange[0]) & (pos <= xrange[1]):
+                return xrange[2]
+
+    def generate_ymaxes(box_plotter, box_names, data_to_ax):
```

Comment:
Although the documentation of the function's role is clear, its code is sometimes a little cryptic. Could you add a few more comments or use more descriptive variable names ? In particular, what do the axes children types mean ? And what does the code do ?